workplace/views.py

```diff
@@ -514,7 +509,33 @@ def get_permissions(self):
     def update(self, request, *args, **kwargs):
         if self.action == "update":
             return Response(status=status.HTTP_405_METHOD_NOT_ALLOWED)
-        return super(ReservationViewSet, self).update(request, *args, **kwargs)
+        response = super(ReservationViewSet, self)\
+            .update(request, *args, **kwargs)
+        if response.status_code == 200 and 'is_present' in request.data:
+            user = User.objects.get(pk=response.data['user_details']['id'])
+            reservation = Reservation.objects.get(pk=response.data['id'])
+            if request.data['is_present']:
+                # user presence set to True: Add tomato for the timeslot
+                Tomato.objects.create(
+                    user=user,
+                    number_of_tomato=reservation.timeslot.number_of_tomatoes,
+                    source=Tomato.TOMATO_SOURCE_TIMESLOT,
+                    content_object=reservation,
+                )
+            else:
+                # User presence set to False: remove tomato for timeslot
+                reservation_type = ContentType.objects.get_for_model(
+                    Reservation)
+                try:
+                    Tomato.objects.get(
+                        user=user,
+                        source=Tomato.TOMATO_SOURCE_TIMESLOT,
+                        content_type=reservation_type,
+                        object_id=reservation.id,
+                    ).delete()
+                except Tomato.DoesNotExist:
+                    pass
```

Comment:
Im afraid that this code became a nightmare to maintain:

1. Should we think about Django Signals to make it automatic from model change ?
2. Should we at least make a subfunction to make sure it's clear what this part of code does  and simplify future contribution ?

Reply:
Logic moved to models and broke down into 1 function